docs/docs/examples/vault.md

```diff
@@ -0,0 +1,264 @@
+
+
+[Vault](https://openbao.org/) is an identity-based secrets and encryption management system, which provides encryption services that are gated by authentication and authorization methods to ensure secure, auditable and restricted access to secrets, such as API keys, passwords, encryption keys or certificates.
+
+Contrast allows to leverage these advantages of having a secure secret and encryption
+management system into a confidential computing environment, further shielding the secrets from the workload operator.
+
+## Sealing and Unsealing of Vaults
+[Sealing](https://openbao.org/docs/concepts/seal/) ensures that all sensitive data within the Vault remains inaccessible and protected when the system isn't in active use.
+It provides a security boundary that prevents unauthorized access during restarts or shutdowns.
+
+Unsealing is required to transition Vault into an operational state, allowing authorized access to stored secrets.
+Vault implementations by default use a set of unseal keys derived from a master key, building up on `Shamir's Secret Sharing` scheme.
+Further to auto-unseal Vaults, the process can be delegated to another already initialized Vault by using
+an exposed [transit secrets engine API](https://openbao.org/api-docs/secret/transit/) as the unsealing mechanism.
+
+
+## Transit secrets engine API of Contrast Coordinator
+To automate the unsealing process in confidential deployments of Vault instances, the coordinator exposes a compatible transit secrets engine API on port 8200.
+Vault deployments can be configured to integrate this transit engine to enable auto-unsealing,
+ensuring immediate operational readiness and seamless integration within the secure Contrast environment.
+
+### Secure endpoints with mutual TLS
+All communication between the transit secrets engine API and Vault is secured through mutual TLS (mTLS), enforced by Contrast’s service mesh.
+Only entities presenting a valid mesh-issued certificate—corresponding to the current state of the Contrast deployment—are trusted.
+
+The Coordinator issues itself a valid certificate at the time of the transit secrets engine API call,
+while the Vault deployment obtains its certificate after successful validation by the mesh API.
+
+### Role of `workloadSecretID`
+To support persistence in the auto-unsealing process, the `workloadSecretID` is used to derive the encryption key utilized by the transit secrets engine. Beyond key derivation, the `workloadSecretID` also plays a critical role in authorization.
+
+Access to a specific encryption key via the transit secrets engine API is permitted only if the requested key name matches the `workloadSecretID` embedded in the corresponding certificate extension of the Contrast mesh certificate. This ensures that each entity is cryptographically bound to its own set of encryption keys within the engine.
```

Comment:
Cryptographically bound? I'd rather say "workloads can only access transit engine secrets for their `workloadSecretID`".